### PR TITLE
PYTHON-3300 Add Explicit Queryable Encryption Example to Docs

### PR DESCRIPTION
**Pull request by juliusgeo**

### CODE REVIEW

File: doc/examples/encryption.rst [hunk 1]
```diff
@@ -409,6 +409,133 @@ Automatic encryption in Queryable Encryption is configured with an ``encrypted_f
...
+Explicit Queryable Encryption (Beta)
+````````````````````````````````````
+
+PyMongo 4.2 brings beta support for Queryable Encryption with MongoDB 6.0.
+
+Queryable Encryption is the second version of Client-Side Field Level Encryption.
+Data is encrypted client-side. Queryable Encryption supports indexed encrypted fields,
+which are further processed server-side.
+
+You must have MongoDB 6.0rc8+ Enterprise to preview the capability.
```
Comment (ShaneHarvey) [inline doc/examples/encryption.rst:421]: Explicit encryption doesn't need the Enterprise server.

File: doc/examples/encryption.rst [hunk 2]
```diff
@@ -409,6 +409,133 @@ Automatic encryption in Queryable Encryption is configured with an ``encrypted_f
...
+                "queryType": "equality"
+              }
+            }
+          ]
+        }
+
+        opts = AutoEncryptionOpts(
+            {"local": {"key": local_master_key}},
+            key_vault.full_name,
+            bypass_query_analysis=True,
```
Comment (ShaneHarvey) [inline doc/examples/encryption.rst:496]: Do we also need bypass_auto_encryption?

File: doc/examples/encryption.rst [hunk 3]
```diff
@@ -409,6 +409,133 @@ Automatic encryption in Queryable Encryption is configured with an ``encrypted_f
...
+              "queries": {
+                "queryType": "equality"
+              }
+            }
+          ]
+        }
+
+        opts = AutoEncryptionOpts(
+            {"local": {"key": local_master_key}},
+            key_vault.full_name,
```
Comment (ShaneHarvey) [inline doc/examples/encryption.rst:495]: Might as well pass `key_vault_client=client` here too

File: doc/examples/encryption.rst [hunk 4]
```diff
@@ -409,6 +409,133 @@ Automatic encryption in Queryable Encryption is configured with an ``encrypted_f
...
+        db = encrypted_client.test
+
+        # Create the collection with encrypted fields.
+        coll = db.create_collection("coll", encrypted_fields=encrypted_fields)
+
+        # Create and encrypt an indexed and unindexed value.
+        val = "encrypted indexed value"
+        unindexed_val = "encrypted unindexed value"
+        insert_payload_indexed = client_encryption.encrypt(val, Algorithm.INDEXED, data_key_id)
+        insert_payload_unindexed = client_encryption.encrypt(unindexed_val, Algorithm.UNINDEXED, data_key_id)
```
Comment (blink1073) [inline doc/examples/encryption.rst:511]: Doesn't this need its own field in `encrypted_fields` and its own unique `data_key_id`?

Comment (ShaneHarvey) [reply doc/examples/encryption.rst]: Great find. Yes, you cannot use the same data_key for two different fields.

File: doc/examples/encryption.rst [hunk 5]
```diff
@@ -409,6 +409,143 @@ Automatic encryption in Queryable Encryption is configured with an ``encrypted_f
...
+            {
+              "keyId": unindexed_key_id,
+              "path": "encryptedUnindexed",
+              "bsonType": "string",
+            }
+          ]
+        }
+        # bypass_auto_encryption=True disable automatic encryption but keeps
+        # the automatic _decryption_ behavior. bypass_auto_encryption will
+        # also disable spawning mongocryptd.
```
Comment (ShaneHarvey) [inline doc/examples/encryption.rst:501]: Given Kevin's explanation in slack and the example here: https://github.com/mongodb/specifications/blob/master/source/client-side-encryption/client-side-encryption.rst#why-is-bypassqueryanalysis-needed we don't want to set bypass_auto_encryption here.

File: doc/examples/encryption.rst [hunk 6]
```diff
@@ -409,6 +409,142 @@ Automatic encryption in Queryable Encryption is configured with an ``encrypted_f
...
+        coll = db.create_collection("coll", encrypted_fields=encrypted_fields)
+
+        # Create and encrypt an indexed and unindexed value.
+        val = "encrypted indexed value"
+        unindexed_val = "encrypted unindexed value"
+        insert_payload_indexed = client_encryption.encrypt(val, Algorithm.INDEXED, indexed_key_id)
+        insert_payload_unindexed = client_encryption.encrypt(unindexed_val, Algorithm.UNINDEXED,
+        unindexed_key_id)
+
+        # Insert the payloads.
```
Comment (blink1073) [inline doc/examples/encryption.rst:522]: payload*

File: doc/examples/encryption.rst [hunk 7]
```diff
@@ -409,6 +409,142 @@ Automatic encryption in Queryable Encryption is configured with an ``encrypted_f
...
+        )
+
+        # Find the document we inserted using the encrypted payload.
+        # The returned document is automatically decrypted.
+        doc = coll.find_one({"encryptedIndexed": find_payload})
+        print('Returned document: %s' % (doc,))
+
+        # Cleanup resources.
+        client_encryption.close()
+        encrypted_client.close()
```
Comment (blink1073) [inline doc/examples/encryption.rst:542]: Need to close the key vault client too.

**Review by blink1073 — CHANGES_REQUESTED** [review]

**Review by ShaneHarvey — APPROVED** [review]

File: doc/examples/encryption.rst (hunk 6 above)
Comment (juliusgeo) [reply doc/examples/encryption.rst:522]: There are technically two distinct payloads.

**Pull request merged**